File: AndroidProject/CollectorAgentMind/src/main/java/com/example/collectoragentmind/codelets/sensors/Touch.java
package com.example.collectoragentmind.codelets.sensors;

import com.example.collectoragentmind.CollectorAgentMind;

import br.unicamp.cst.core.entities.Codelet;

public class Touch extends Codelet {

    /*
        Requests the apples in reach every proc. The body
        should respond to this request by updating the
        touchMO through the plugin's interface
    */

    private CollectorAgentMind plugin;

    public Touch() {
        plugin = CollectorAgentMind.getInstance();
    }

    @Override
    public void accessMemoryObjects() {

    }

    @Override
    public void calculateActivation() {

    }

    @Override
    public void proc() {
        plugin.emitRequest("request_apples_in_reach");
    }
}

File: GodotKotlin/CoreModelTrail/src/main/kotlin/com/example/game/cst/codelets/sensors/Touch.kt
package com.example.game.cst.codelets.sensors

import br.unicamp.cst.core.entities.Codelet
import br.unicamp.cst.core.entities.MemoryObject
import com.example.game.CollectorMindCommunicator


class Touch : Codelet() {
    /*
        Requests the apples in reach every proc. The body
        should respond to this request by updating the
        touchMO through the plugin's interface
    */

    private var touchMO: MemoryObject? = null

    override fun accessMemoryObjects() {
        touchMO = getInput("TOUCH") as MemoryObject?
    }

    override fun calculateActivation() {}

    override fun proc() {
        touchMO?.setI(CollectorMindCommunicator.instance.getApplesInReach())
    }
}
